src/scanner.cc

```diff
@@ -505,70 +524,47 @@ class Scanner
      */
     bool parse_text(TSLexer* lexer)
     {
-        while (lexer->lookahead)
+        if (!m_TagStack.empty())
         {
-            if (m_TagStack.size() > 0 && m_IndentationLevel < m_TagStack.back())
-            {
-                lexer->result_symbol = RANGED_TAG_END;
-                return true;
-            }
+            while (lexer->lookahead && lexer->lookahead != '\n')
+                advance(lexer);
 
-            // If we have an escape sequence in the middle of the paragraph then terminate the paragraph
-            // to allow the escape sequence to get parsed
-            if (lexer->lookahead == '\\')
-            {
-                lexer->mark_end(lexer);
+            lexer->result_symbol = m_LastToken = WORD;
+            return true;
+        }
 
+        if (lexer->lookahead == ' ' || lexer->lookahead == '\t')
+        {
+            do
                 advance(lexer);
+            while (lexer->lookahead && (lexer->lookahead == ' ' || lexer->lookahead == '\t'));
 
-                if (!lexer->lookahead)
-                    continue;
+            lexer->result_symbol = m_LastToken = SPACE;
+            return true;
+        }
 
-                m_LastToken = PARAGRAPH_SEGMENT;
-                lexer->result_symbol = PARAGRAPH_SEGMENT;
-                return true;
-            }
+        const TokenType resulting_symbol = (bool)std::iswupper(lexer->lookahead) ? CAPITALIZED_WORD : WORD;
 
-            // Try and find an occurrence of a trailing modifier
-            if (!std::iswspace(m_Current) && lexer->lookahead == '~')
+        do
+        {
+            if (lexer->lookahead == '~')
             {
                 advance(lexer);
 
-                // If we've managed to find one then skip over the newline and continue parsing
                 if (lexer->lookahead == '\n')
-                    continue;
-            }
-            // A [ is a special symbol - it can both mean a todo item and a link
-            // Halt the parsing of the paragraph segment if such a thing is encountered
-            else if (m_TagStack.size() == 0 && (std::iswspace(m_Current) || std::iswpunct(m_Current)) && lexer->lookahead == '[')
-            {
-                m_LastToken = PARAGRAPH_SEGMENT;
-                lexer->result_symbol = PARAGRAPH_SEGMENT;
-                return true;
-            }
-
-            advance(lexer);
+                {
+                    advance(lexer);
 
-            if (lexer->lookahead == '\n')
-                break;
+                    if (!lexer->lookahead || lexer->eof(lexer))
+                        return false;
+                }
+            }
+            else
+                advance(lexer);
         }
+        while (lexer->lookahead && (lexer->lookahead != '\n' && lexer->lookahead != ' ' && lexer->lookahead != '\t')); // TODO: Perform specific checks for attached modifiers
```

Comment:
`\r` and `\f` are intentionally skipped?
If not then maybe check for `std::isspace`?
```suggestion
        while (lexer->lookahead && std::isspace(lexer->lookahead)); // TODO: Perform specific checks for attached modifiers
```

Reply:
Right. All of the things where we explicitly check for ` ` or `\t` were actually on purpose. I don't remember whether the issue was fixed but using std::iswspace messed up something... somewhere. I'll try reverting it since I don't exactly remember what bugs it caused (and whether or not those bugs are still there)